bin/xbps-remove/clean-cache.c

```diff
@@ -32,10 +32,84 @@
 #include <assert.h>
 #include <unistd.h>
 #include <dirent.h>
+#include <glob.h>
 
 #include <xbps.h>
 #include "defs.h"
 
+static int keep_cache;
+
+static bool
+is_removable_pkg(struct xbps_handle *xhp, const char *binpkg, char* pkgver)
+{
+
+	char pkgname[XBPS_NAME_SIZE];
+	xbps_string_t pattern, pkg;
+	glob_t fake_results;
+	int len, rv;
+	const char *entry = NULL;
+	xbps_array_t real_results = NULL;
+	bool match = false;
+	xbps_object_t obj = NULL;
+
+	len = rv = 0;
+	pattern = pkg = NULL;
+
+	if (xbps_pkg_name(pkgname, sizeof(pkgname), pkgver)) {
+		len = strlen(pkgname);
+		/* Building patterns */
+		pattern = xbps_string_create_cstring(pkgname);
+		assert(pattern);
+		xbps_string_append_cstring(pattern, "-");
+		xbps_string_append_cstring(pattern, "*.xbps");
+
+		assert(glob(xbps_string_cstring_nocopy(pattern), 0, NULL, &fake_results) == 0);
```

Comment:
You shouldn't have expressions with side effects inside assertions.

Reply:
That glob() function must forcely returned something. For this reason i assert that. Anyway, i can handle the result in different way.